### ISSUE TITLE
train loss 很大

### ISSUE THREAD

**Issue by 315654547** [Ucas-HaoranWei/GOT-OCR2.0#118]
用readme里面给的训练脚本：
deepspeed   GOT/train/train_GOT.py \
 --deepspeed zero_config/zero2.json    --model_name_or_path ./GOT_weights/ \
 --use_im_start_end True   \
 --bf16 True   \
 --gradient_accumulation_steps 2    \
 --evaluation_strategy "no"   \
 --save_strategy "steps"  \
 --save_steps 200   \
 --save_total_limit 1   \
 --weight_decay 0.    \
 --warmup_ratio 0.001     \
 --lr_scheduler_type "cosine"    \
 --logging_steps 1    \
 --tf32 True     \
 --model_max_length 8192    \
 --gradient_checkpointing True   \
 --dataloader_num_workers 8    \
 --report_to none  \
 --per_device_train_batch_size 3    \
 --num_train_epochs 1  \
 --learning_rate 2e-5   \
 --datasets scene \
 --output_dir ./ckpt/

1.把GOT_ocr_2_0.py中的vision model改成with torch.set_grad_enabled(True)

2.train_GOT中用trainer_vit_fixlr的GOTTrainer训练

训练loss初始为：
{'loss': 3.7446, 'learning_rate': 1.999047561485943e-05, 'epoch': 0.01}                                  
{'loss': 4.298, 'learning_rate': 1.999044389841871e-05, 'epoch': 0.01}                                   
{'loss': 4.606, 'learning_rate': 1.9990412129282704e-05, 'epoch': 0.01}                                  
{'loss': 4.2281, 'learning_rate': 1.9990380307451584e-05, 'epoch': 0.01}                                 
{'loss': 4.2816, 'learning_rate': 1.999034843292552e-05, 'epoch': 0.01} 

问题1:
请问为啥初始loss这么大？一般finetune的初始loss不是很小吗？

问题2:
train文件下的那几个GOTTrainer的区别是啥？比如trainer_llm_llrd、trainer_vit_fixlr、trainer_vit_llrd。

**Comment by Ucas-HaoranWei** [comment]
loss应该非常小，你这个应该有问题，看看你的真值json

**Comment by 315654547** [comment]
{
        "id": "/mnt/data/Next-Generation-OCR/dataset/src/screenshot_train/images/65587.jpg",
        "task": "text_reading",
        "dataset_name": "screenshot_train",
        "images": [
            "screenshot_train/images/65587.jpg"
        ],
        "conversations": [
            {
                "from": "human",
                "value": "<image>\nRead all the text in the image."
            },
            {
                "from": "gpt",
                "value": "f o ① * 4：28 寻 Name or Phone"
            }
        ],
        "question_id": 1
    }

看起来数据没有什么问题。

**Comment by 315654547** [comment]
"from": "human",
"value": "<image>\nRead all the text in the image."
粘贴的有点问题，我把Read all the text in the image也换成了OCR，跟给的示例完全一样。

**Comment by Ucas-HaoranWei** [comment]
知道你的问题了，没有<image>，你当成nlp训了，"<image>\nRead all the text in the image."

**Comment by 315654547** [comment]
{
        "id": "/mnt/data/Next-Generation-OCR/dataset/src/screenshot_train/images/65587.jpg",
        "task": "text_reading",
        "dataset_name": "screenshot_train",
        "images": [
            "screenshot_train/images/65587.jpg"
        ],
        "conversations": [
            {
                "from": "human",
                "value": "<image>\nOCR."
            },
            {
                "from": "gpt",
                "value": "f o ① * 4：28 寻 Name or Phone"
            }
        ],
        "question_id": 1
    },

不是nlp，是ocr

**Comment by Ucas-HaoranWei** [comment]
![IMG_5582](https://github.com/user-attachments/assets/0102e387-d9b8-4904-80f3-3ccb1ccaf67c)
不好意思，前端吞了我的输入

**Comment by 315654547** [comment]
解决了，images改成image，训练loss正常。
第2个问题：
train文件下的那几个GOTTrainer的区别是啥？比如trainer_llm_llrd、trainer_vit_fixlr、trainer_vit_llrd。

如果我想把前面的vision encoder换成大一点的vit模型，具体训练顺序是什么样子的？

**Comment by Ucas-HaoranWei** [comment]
额，应该还不对，因为你human-value 没有`<image>`字段，`<image>\nOCR:`
trainer_llm_llrd、trainer_vit_fixlr、trainer_vit_llrd 这是调前后不同学习率的可以忽略
另外最重要的就是vision encoder，你可以随意换llm，但是换encoder很难work

**Comment by 315654547** [comment]
粘贴的时候系统把image弄没了，实际是有的，loss是正常的。

要换vision encoder是不是得按照论文所说的，按照3个步骤全部走一遍，得需要大量的数据支撑？
